src/coreclr/vm/field.cpp

```diff
@@ -140,7 +143,8 @@ TypeHandle FieldDesc::LookupFieldTypeHandle(ClassLoadLevel level, BOOL dropGener
 
     // Caller should have handled all the non-class cases, already.
     _ASSERTE(GetFieldType() == ELEMENT_TYPE_CLASS ||
-             GetFieldType() == ELEMENT_TYPE_VALUETYPE);
+             GetFieldType() == ELEMENT_TYPE_VALUETYPE ||
+             GetFieldType() == ELEMENT_TYPE_TYPEDBYREF);
```

Comment:
Can `GetFieldType()` ever return `ELEMENT_TYPE_TYPEDBYREF` given the adjustment in Init?

Reply:
Nope, it shouldn't. Will update.